BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,234 @@
+"""
+The current tooling supported in CMake, Homebrew, and QT5 are insufficient for
+creating MacOSX universal binaries automatically for applications like Dolphin
+which have more complicated build requirements (like different libraries, build
+flags and source files for each target architecture).
+
+So instead, this script manages the configuration and compilation of distinct
+builds and project files for each target architecture and then merges the two
+binaries into a single universal binary.
+
+Running this script will:
+1) Generate Xcode project files for the ARM build (if project files don't
+   already exist)
+2) Generate Xcode project files for the x64 build (if project files don't
+   already exist)
+3) Build the ARM project for the selected build_target
+4) Build the x64 project for the selected build_target
+5) Generates universal .app packages combining the ARM and x64 packages
+6) Utilizes the lipo tool to combine the binary objects inside each of the
+   packages into universal binaries
+7) Code signs the final universal binaries using the specified
+   codesign_identity
+"""
+
+import glob
+import sys
+import os
+import shutil
+import filecmp
+import argparse
+import subprocess
+
+# #BEGIN CONFIG# #
+
+# The config variables listed below are the defaults, but they can be
+# overridden by command line arguments see parse_args(), or run:
+# BuildMacOSUniversalBinary.py --help
+
+
+# Location of destination universal binary
+dst_app = "universal/"
+# Build Target (dolphin-emu to just build the emulator and skip the tests)
+build_target = "ALL_BUILD"
+
+# Locations to pkg config files for arm and x64 libraries
+# The default values of these paths are taken from the default
+# paths used for homebrew
+pkg_config_path = {
+    "arm64":  '/opt/homebrew/lib/pkgconfig',
+    "x86_64": '/usr/local/lib/pkgconfig'
+}
+
+# Locations to qt5 directories for arm and x64 libraries
+# The default values of these paths are taken from the default
+# paths used for homebrew
+qt5_path = {
+    "arm64":  '/opt/homebrew/opt/qt5',
+    "x86_64": '/usr/local/opt/qt5'
+}
+
+# Identity to use for code signing. "-" indicates that the app will not
+# be cryptographically signed/notarized but will instead just use a
+# SHA checksum to verify the integrity of the app. This doesn't
+# protect against malicious actors, but it does protect against
+# running corrupted binaries and allows for access to the extended
+# permisions needed for ARM builds
+
+codesign_identity = '-'
+
+# # END CONFIG # #
+
+# Architectures to build for. This is explicity left out of the command line
+# config options for several reasons:
+# 1) Adding new architectures will generally require more code changes
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+architectures = ["x86_64", "arm64"]
+
+# Minimum macOS version for each architecture slice
+mac_os_deployment_target={
+    "arm64": "10.14.0",
```

Comment:
It makes little sense to target anything below 11.0 for Apple Silicon.

Reply:
Okay, I changed the target for arm64 to 11.0.0